test/sd-test-srv/main.go

```diff
@@ -12,6 +12,24 @@ import (
 	"github.com/miekg/dns"
 )
 
+// proxyQuery takes a query for a domain and proxies it to Cloudflare DNS.
+// This allows us to shim in sd-test-srv for our container without blocking
+// lookups to github.com and coveralls.io, which are used in our `godep-restore`
+// and `coverage` test phases respectively.
+func proxyQuery(w dns.ResponseWriter, r *dns.Msg) {
+	c := new(dns.Client)
+	in, _, err := c.Exchange(r, "1.1.1.1:53")
```

Comment:
I'd prefer we not hard code a specific upstream resolver to use and instead defer to the configured system resolver. That said we can't just assume 127.0.0.1 works so we'd have to use something like `miekg/dns. ClientConfigFromFile` to read the /etc/resolv.conf file to determine where to go.

I'm not super against using this specific resolver per-se, it just seems like a bit of an anti-pattern. If you think this is too much extra work I can probably be convinced it's not worth it.

Reply:
Yeah, I wanted to use the system resolver, but in Docker that's 127.0.1.1, which is Docker's embedded DNS, which will pass along unknown queries to sd-test-srv, which will pass along unknown queries to sd-test-srv... :-D